seed/utils/search.py

```diff
@@ -507,3 +507,40 @@ def build_view_filters_and_sorts(
             annotations.update(parsed_annotations)
 
     return new_filters, annotations, order_by
+
+
+def build_related_model_filters_and_sorts(filters: QueryDict, columns: list[dict]) -> tuple[Q, AnnotationDict, list[str]]:
+    order_by = []
+    annotations = {}
+    columns_by_name = {}
+    for column in columns:
+        if column["related"]:
+            continue
+        columns_by_name[column["name"]] = column
+
+    column_name = filters.get("order_by")
+    if not column_name:
+        return Q(), {}, ["id"]
+
+    direction = "-" if column_name.startswith("-") else ""
+    column_name = column_name.lstrip("-")
+
+    if "goal_note" in column_name:
+        column_name = column_name.replace("goal_note", "goalnote")
+
+    boolean_column = column_name == "property__goalnote__passed_checks"
+    target: Union[bool, str]
+    if boolean_column:
+        target = False
+    else:
+        target = ""
+
+    related_model = Case(When(**{column_name: target}, then=Value(1)), default=Value(0), output_field=IntegerField())
```

Comment:
By defautl, django will follow these patterns
- asc: None, "A", "B", "Z"
- desc: "Z", "B", "A", None
I thought this was counterintuitive and if a user sorted a text column like historical notes, "A" should be at the top, and blanks should below "Z". If this doesnt matter I can simplify this function by removing the `related_model` annotation

Reply:
I like the way you implemented it!